src/binary/ibc-setup/commands/connections.spec.ts

```diff
@@ -0,0 +1,68 @@
+import fs from 'fs';
+
+import test from 'ava';
+import sinon from 'sinon';
+import { Logger } from 'winston';
+
+import { Link } from '../../../lib/link';
+import { TestLogger } from '../../../lib/testutils';
+import { signingClient } from '../../utils/signing-client';
+
+import { simappChain, wasmdChain } from './chains';
+import { Options, run } from './connections';
+
+const fsReadFileSync = sinon.stub(fs, 'readFileSync');
+const consoleLog = sinon.stub(console, 'log');
+
+const mnemonic =
+  'enlist hip relief stomach skate base shallow young switch frequent cry park';
+
+const registryYaml = `
+version: 1
+
+chains:
+  local_wasm:
+    chain_id: testing
+    prefix: wasm
+    gas_price: 0.025ucosm
+    rpc:
+      - http://localhost:26659
+  local_simapp:
+    chain_id: simd-testing
+    prefix: cosmos
+    gas_price: 0.025umuon
+    rpc:
+      - http://localhost:26658`;
+
+test.beforeEach(() => {
+  sinon.reset();
+});
+
+test.serial('lists connections', async (t) => {
+  const logger = new TestLogger();
+
+  const ibcClientSimapp = await signingClient(simappChain, mnemonic);
+  const ibcClientWasm = await signingClient(wasmdChain, mnemonic);
+
+  const link = await Link.createWithNewConnections(
+    ibcClientSimapp,
+    ibcClientWasm
+  );
+
+  const options: Options = {
+    home: '/home/user',
+    mnemonic,
+    chain: 'local_simapp',
+  };
+
+  fsReadFileSync.returns(registryYaml);
+
+  await run(options, (logger as unknown) as Logger);
+
+  const match = new RegExp(`${link.endA.connectionID}\\s+Open`);
+  t.assert(consoleLog.getCall(-1).calledWithMatch(match));
+});
+
+// test.serial('logs a message when no connections are found', async (t) => {
+//   //
+// });
```

Comment:
I would like to write a test for the `no connections found` case, but all tests operate on the same chains, and the connections are already created.

I thought of solutions:
- somehow force this kind of tests to run in the very beginning when the chains are fresh
- create a new chain for this kind of tests

I wouldn't say I like any of those. Maybe I'm missing something, and maybe testing this is not worth at all. Thoughts? @ethanfrey @willclarktech

Reply:
A fresh chain with no connections sounds sensible to me.

Reply:
Compare with our test setup in CosmJS, where we have a number of preset accounts including one that's unused: https://github.com/cosmos/cosmjs/blob/main/scripts/wasmd/README.md#preset-accounts

Reply:
How the preset accounts are related to this? I think listing the connections is account agnostic so we would have to setup a new chain.


Another solution I came up with:
- stub ibc client (don't test against running blockchain).

Reply:
> How the preset accounts are related to this?

I just present them as an analogy: in that case we need accounts as test fixtures and we include accounts that aren't used to help test certain cases. In this project we need chains as test fixtures and including chains with no connections seems equally helpful.

Reply:
Okay, there is a 3-chain test setup ticket, but assuming they all have connections. Please add another issue for "Add simapp chain to CI with no IBC connections" and then mention adding more test cases here when it was added. I would not block merging on this, just note a follow-up.

You could also manually test (once) by starting new CI chains and just running this one test file, with `test.only()` on this test case.

Reply:
I created [a new issue](https://github.com/confio/ts-relayer/issues/130) and mentioned it in the comment: e2304ce